Sources/CryptoBoringWrapper/EC/EllipticCurvePoint.swift

```diff
@@ -12,41 +12,262 @@
+    package static func subtracting(_ rhs: EllipticCurvePoint, from lhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) throws -> EllipticCurvePoint {
+        try EllipticCurvePoint(subtracting: rhs, from: lhs, on: group)
+    }
+
+    @usableFromInline
+    package init<MessageBytes: ContiguousBytes, DSTBytes: ContiguousBytes>(hashing msg: MessageBytes, to group: BoringSSLEllipticCurveGroup, domainSeparationTag: DSTBytes) throws {
+        let hashToCurveFunction = switch group.curveName {
+        case .p256: CCryptoBoringSSLShims_EC_hash_to_curve_p256_xmd_sha256_sswu
+        case .p384: CCryptoBoringSSLShims_EC_hash_to_curve_p384_xmd_sha384_sswu
+        case .p521: throw CryptoBoringWrapperError.invalidParameter // BoringSSL doesn't have a hash_to_curve API for P521.
+        case .none: throw CryptoBoringWrapperError.internalBoringSSLError()
+        }
+
+       self._basePoint = try msg.withUnsafeBytes { msgPtr in
+            try group.withUnsafeGroupPointer { groupPtr in
+                try domainSeparationTag.withUnsafeBytes { dstPtr in
+                    guard
+                        let pointPtr = CCryptoBoringSSL_EC_POINT_new(groupPtr),
+                        hashToCurveFunction(
+                            groupPtr,
+                            pointPtr,
+                            dstPtr.baseAddress,
+                            dstPtr.count,
+                            msgPtr.baseAddress,
+                            msgPtr.count
+                        ) == 1
+                    else { throw CryptoBoringWrapperError.internalBoringSSLError() }
+                    return pointPtr
+                }
+            }
+        }
+    }
+
+    @usableFromInline
+    package func isEqual(to rhs: EllipticCurvePoint, on group: BoringSSLEllipticCurveGroup) -> Bool {
+        self.withPointPointer { selfPtr in
+            group.withUnsafeGroupPointer { groupPtr in
+                rhs.withPointPointer { rhsPtr in
+                    switch CCryptoBoringSSL_EC_POINT_cmp(groupPtr, selfPtr, rhsPtr, nil) {
+                    case 0: return true
+                    case 1: return false
+                    default: fatalError("Unexpected error testing EC point equality")
+                    }
+                }
+            }
+        }
+    }
+
+    @usableFromInline
+    package init<Bytes: ContiguousBytes>(x962Representation bytes: Bytes, on group: BoringSSLEllipticCurveGroup) throws {
         self._basePoint = try group.withUnsafeGroupPointer { groupPtr in
-            guard let basePoint = CCryptoBoringSSL_EC_POINT_dup(pointer, groupPtr) else {
+            guard let basePoint = CCryptoBoringSSL_EC_POINT_new(groupPtr) else {
                 throw CryptoBoringWrapperError.internalBoringSSLError()
             }
             return basePoint
         }
+
+        guard group.withUnsafeGroupPointer({ groupPtr in
+            bytes.withUnsafeBytes { dataPtr in
+                CCryptoBoringSSL_EC_POINT_oct2point(
```

Comment:
If this fails we leak the point.

Reply:
Resolved in the same manner as https://github.com/apple/swift-crypto/pull/292/files/fda1a2c6df294e87b24b77ad28a7cd5950d74d25#r1850181741.